starcli/search.py

```diff
@@ -211,65 +212,17 @@ def search_github_trending(
     language=None, spoken_language=None, order="desc", stars=">=10", date_range=None
 ):
     """ Returns trending repositories from github trending page """
-    url = "https://github.com/trending?"  # filter for spoken language is available only here
-    if language:
-        url += f"language={language}"  # filter by programming language
-    if spoken_language:
-        url += f"&spoken_language_code={spoken_language}"  # filter by spoken language
+    gtrending_repo_list = []
     if date_range:
-        url += f"&since={date_range_map[date_range]}"
-
-    request = get_valid_request(url)
-    if request is None:
-        return request
-
-    page = request.text
-
-    soup = BeautifulSoup(page, "lxml")
-    repo_list = soup.find_all("article", class_="Box-row")
-    repositories = []
-    for repo in repo_list:
-        repo_dict = {}
-        repo_dict["full_name"] = repo.h1.a.text.replace(" ", "").replace("\n", "")
-        repo_dict["name"] = repo_dict["full_name"].split("/")[1]
-        repo_dict["html_url"] = f"https://github.com/{repo_dict['full_name']}"
-        div = repo.find("div", class_="f6 text-gray mt-2")
-        anchor_list = div.find_all("a")
-        span_list = div.find_all("span")
-        # if stars present
-        repo_dict["stargazers_count"] = (
-            int(anchor_list[0].text.strip().replace(",", ""))
-            if anchor_list[0].text
-            else -1
-        )
-        # if forks count present
-        repo_dict["forks_count"] = (
-            int(anchor_list[1].text.strip().replace(",", ""))
-            if anchor_list[1].text
-            else -1
+        gtrending_repo_list = fetch_repos(
+            language, spoken_language, date_range_map[date_range]
         )
-        # if language is present
-        language_span = div.span.find_all("span")
-        if len(language_span) > 0:
-            if (
-                not language or language_span[1].text.strip().lower() == language
-            ):  # if major lang of repo is same as language
-                repo_dict["language"] = language_span[1].text.strip()
-            else:
-                repo_dict["language"] = language
-        else:
-            repo_dict["language"] = None
-        # if description is present
-        repo_dict["description"] = repo.p.text.strip() if repo.p else None
-        # if stars date range is present
-        if date_range:
-            repo_dict["date_range"] = (
-                span_list[-1].text.replace("\n", "").strip()
-                if len(span_list) > 0
-                else None
-            )
-        else:
-            repo_dict["date_range"] = None
+    else:
+        gtrending_repo_list = fetch_repos(language, spoken_language)
+    repositories = []
+    for gtrending_repo in gtrending_repo_list:
+        repo_dict = convert_gtrending_repo_to_repo(gtrending_repo)
+        repo_dict["date_range"] = date_range_map[date_range] if date_range else None
```

Comment:
The value of `repo_dict["date_range"]` shouldn't be daily, monthly, weekly, it should be `12 stars today`, `394 stars this week` etc, because this assigning this to `date_range_map[date_range]` can lead to in `__main__.py` line number 152 
https://github.com/hedythedev/starcli/blob/6c0cd6db4f2ed9adc28ef1d264edc497d1ed8d67/starcli/__main__.py#L152
throwing

```
ValueError: invalid literal for int() with base 10: 'daily'
```

To solve this problem, we can put the assignment of date_range in the gtrending repo dict conversion function instead, and assign `repo["date_range"]` to `gtrending_repo.get("currentPeriodStars")`, then on this line here, use
```python
repo_dict["date_range"] = repo_dict["date_range"] + " stars " + date_range
```

so that the value of `repo_dict["date_range"]` is something like '12 stars today'